plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hms/TrinoHiveCatalog.java

```diff
@@ -261,6 +268,38 @@ public Transaction newCreateTableTransaction(
                 isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser()));
     }
 
+    @Override
+    public void registerTable(ConnectorSession session, SchemaTableName schemaTableName, String metadataLocation)
+            throws TrinoException
+    {
+        if (tableExists(schemaTableName)) {
+            throw new TrinoException(TABLE_ALREADY_EXISTS, format("Table '%s' already exists in schema '%s'", schemaTableName.getTableName(), schemaTableName.getSchemaName()));
+        }
+
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+        TableMetadata tableMetadata = readMetadata(fileSystem, metadataLocation);
+        Optional<String> owner = isUsingSystemSecurity ? Optional.empty() : Optional.of(session.getUser());
+
+        io.trino.plugin.hive.metastore.Table table = makeTable(tableMetadata, schemaTableName.getSchemaName(), schemaTableName.getTableName(), owner, metadataLocation);
+        PrincipalPrivileges privileges = owner.map(MetastoreUtil::buildInitialPrivilegeSet).orElse(NO_PRIVILEGES);
+        metastore.createTable(table, privileges);
+    }
+
+    private boolean tableExists(SchemaTableName schemaTableName)
+    {
+        return metastore.getTable(schemaTableName.getSchemaName(), schemaTableName.getTableName()).isPresent();
+    }
+
+    /**
+     * Read metadata from provided metadata file location. This API should be called for valid metadata location.
+     */
+    private static TableMetadata readMetadata(TrinoFileSystem fileSystem, String metadataLocation)
```

Comment:
Why does the Hive version read the TableMetadata but Glue does not?

Reply:
I was following older implementation of `AbstractMetastoreTableOperations#commitNewTable` where we require to have `tableMetadata`. After your comment, I realized that we don't need `tableMetadata` instead with `metadataLocation` we can actually create table. Updated the code with new change. Thanks for pointing this out.